flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -887,7 +887,8 @@ public void receiveDeclineMessage(DeclineCheckpoint message, String taskManagerL
 					checkpointId,
 					message.getTaskExecutionId(),
 					job,
-					taskManagerLocationInfo);
+					taskManagerLocationInfo,
+					message.getReason());
```

Comment:
Isn't this is missing a pattern/format change? Also how would you like it to be logged? Just the `message.getReason().toString()`? Do we care about the stack trace?

Reply:
No, this is a vararg, and the last argument is interpreted as `Throwable`.
The stacktrace is printed as I showed in the description.

Reply:
Good to know, I was not aware of that.